src/java.base/share/native/libjli/args.c

```diff
@@ -354,42 +354,38 @@ static JLI_List readArgFile(FILE *file) {
     return rv;
 }
 
+static void reportAndExit(const char* fmt, const char* arg) {
+    if (fmt != NULL) JLI_ReportMessage(fmt, arg);
```

Comment:
the if (fmt != NULL) check wouldn't be necessary here if you fix the other location, I think